README.md

```diff
@@ -1,4 +1,92 @@
-# prestige
-A state management architecture for Swift inspired by Vuex 
+# Prestige
 
-Coming Soon!
+Prestige is a predictable state container, written in Swift. Inspired by [Redux](http://redux.js.org) and [Vuex](https://vuex.vuejs.org), it aims to enforce separation of concerns and a unidirectional data flow by keeping your entire app state in a single data structure that shouldn't be mutated directly, instead relying on an action dispatch mechanism to describe changes. Prestige uses [PromiseKit](https://github.com/mxcl/PromiseKit) for whenever a dispatch is done.
+
+## Usage
+
+Define your app **state**. In this example I am using a `struct`. the state of your application is stored in a single data structure. This makes it easier to understand the behavior of the app state at any given point in time, simplifies state persistence and improves code readability.:
+
+```swift
+struct AppState: State {
+    let counter: Int
+}
+```
+
+As an example, operations that users might perform within your app would be described as **actions**. State should only be mutated through dispatched **actions**, lightweight objects that mutate the state. Since mutations are centralized, inconsistencies are infrequent and race-conditions become easier to avoid.:
+
+```swift
+struct IncrementAction {
+    let increment: Int = 0
+}
+
+extension IncrementAction: Action {
+   typealias State = AppState
+  
+    func mutate(_ state: AppState) -> Promise<AppState> {
+        let count = state.counter + 1
+        return .value(AppState(counter: count))
```

Comment:
@Prince2k3 In the example, it might be good to replace `let count = state.counter + 1` with `let count = state.counter + increment`. That way, people can more easily tie back to the benefit of defining properties for the Action struct.

Reply:
doh that is true lol